### ISSUE TITLE
Runnung Ansible tasks as specific user Inside a Docker container fails to transfer/chown the module code correctly.

### ISSUE THREAD

**Issue by Bazilbrush** [hashicorp/packer#8502]
#### Overview of the Issue

I am building a docker jenkins slave while reusing Ansible playbooks to provision the required components. When trying to run git_config ansible module on the target container, I switch to use 'jenkins' user to set up its git environment. This task fails on chown step of module code provisioning.

#### Reproduction Steps
Packer config json:
```
{
    "builders": [{
        "type": "docker",
        "image": "centos:7",
        "commit": true
        
    }],
    "provisioners": [
        
        {
        "type": "ansible",
        "playbook_file": "src/main/resources/ansible/provision.yml",
        "user": "root",
        "extra_arguments": [
            "--extra-vars", "@src/main/resources/ansible/vars/production.yml"
           
            ]
        }
    ],
    "post-processors": [[ {
        "type": "docker-tag",
        "repository": "docker-slave",
        "tag": "latest"
    } ]]
}
```
Ansible task in question:
```
- name: Set git user name
  become: yes
  become_user: "{{ jenkins_username }}"
  git_config:
    name: user.name
    scope: global
    value: SonOfJenkins
```

### Packer version
1.4.5
### Ansible version 
2.8.1
### Docker version
1.13.1

### Operating system and Environment details
Fedora 29, kernel 5.3.11-100.fc29.x86_64

### Log Fragments and crash.log files

```docker: TASK [configure/git : Set git user name] ***************************************
2019/12/18 16:23:13 packr: 2019/12/18 16:23:13 new env request: LANG=en_GB.UTF-8
2019/12/18 16:23:13 packr: 2019/12/18 16:23:13 new exec request: /bin/sh -c '( umask 77 && mkdir -p "` echo /var/tmp/ansible-tmp-1576686193.7901585-202806289542259 `" && echo ansible-tmp-1576686193.7901585-202806289542259="` echo /var/tmp/ansible-tmp-1576686193.7901585-202806289542259 `" ) && sleep 0'
2019/12/18 16:23:13 packr: 2019/12/18 16:23:13 Executing docker exec -i e131d8cb91211b4bc2a51b2ea16d0c55fa2d81d891194a5b84b9cbadf78bf53e /bin/sh -c (/bin/sh -c '( umask 77 && mkdir -p "` echo /var/tmp/ansible-tmp-1576686193.7901585-202806289542259 `" && echo ansible-tmp-1576686193.7901585-202806289542259="` echo /var/tmp/ansible-tmp-1576686193.7901585-202806289542259 `" ) && sleep 0'):
2019/12/18 16:23:13 packr: 2019/12/18 16:23:13 [INFO] RPC endpoint: Communicator ended with: 0
2019/12/18 16:23:13 [INFO] 103 bytes written for 'stdout'
2019/12/18 16:23:13 [INFO] 0 bytes written for 'stderr'
2019/12/18 16:23:13 [INFO] RPC client: Communicator ended with: 0
2019/12/18 16:23:13 [INFO] RPC endpoint: Communicator ended with: 0
2019/12/18 16:23:13 [INFO] 0 bytes written for 'stdin'
2019/12/18 16:23:13 packr: 2019/12/18 16:23:13 [INFO] 103 bytes written for 'stdout'
2019/12/18 16:23:13 packr: 2019/12/18 16:23:13 [INFO] 0 bytes written for 'stderr'
2019/12/18 16:23:13 packr: 2019/12/18 16:23:13 [INFO] RPC client: Communicator ended with: 0
2019/12/18 16:23:13 packr: 2019/12/18 16:23:13 [INFO] 0 bytes written for 'stdin'
2019/12/18 16:23:14 packr: 2019/12/18 16:23:14 new env request: LANG=en_GB.UTF-8
2019/12/18 16:23:14 packr: 2019/12/18 16:23:14 new exec request: scp -t /var/tmp/ansible-tmp-1576686193.7901585-202806289542259/AnsiballZ_git_config.py
2019/12/18 16:23:14 packr: 2019/12/18 16:23:14 Copying to /var/tmp/ansible-tmp-1576686193.7901585-202806289542259/AnsiballZ_git_config.py on container e131d8cb91211b4bc2a51b2ea16d0c55fa2d81d891194a5b84b9cbadf78bf53e.
2019/12/18 16:23:14 [INFO] 103317 bytes written for 'uploadData'
2019/12/18 16:23:14 packr: 2019/12/18 16:23:14 [INFO] 103317 bytes written for 'uploadData'
2019/12/18 16:23:14 packr: 2019/12/18 16:23:14 Copied 103317 bytes for /var/tmp/ansible-tmp-1576686193.7901585-202806289542259/AnsiballZ_git_config.py
2019/12/18 16:23:14 packr: 2019/12/18 16:23:14 Failed to set owner of the uploaded file: exit status 1, chown: cannot access '/var/tmp/ansible-tmp-1576686193.7901585-202806289542259/AnsiballZ_git_config.py': No such file or directory
2019/12/18 16:23:14 packr: 
    docker: fatal: [default]: FAILED! =>
    docker:   msg: |-
    docker:     failed to transfer file to /home/jbazileuski/.ansible/tmp/ansible-local-19447c22y5col/tmpa86pjxrs /var/tmp/ansible-tmp-1576686193.7901585-202806289542259/AnsiballZ_git_config.py:
    docker:
    docker: PLAY RECAP *********************************************************************
    docker: default                    : ok=76   changed=58   unreachable=0    failed=1    skipped=1    rescued=0    ignored=0
    docker
```

**Comment by nywilken** [comment]
Hi @Bazilbrush this seems like an Ansible issue, not a Packer issue. I'd recommend reaching out to the [mailing list](https://groups.google.com/forum/#!forum/packer-tool) or the [community page](https://learn.hashicorp.com/packer/). If after reaching out to the mailing list you still think this is a Packer issue gladly comment on this issue and I will reopen it.

**Comment by nywilken** [comment]
Apologies, I closed without adding any context. I tried reproducing the issue using the provided configuration but did not receive any issues. I had to install sudo and other tools onto the container to get things to work, but I suspect your playbook maybe handling all of that as well. I did notice the file not found error in your log so I would check that the filename is correct before trying again. Again, if you reach out on the mailing list folks with a bit more Ansible knowledge might be able to better assist.
